daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/processors/unparsers/ChoiceAndOtherVariousUnparsers.scala

```diff
@@ -43,7 +44,11 @@ class ComplexTypeUnparser(rd: RuntimeData, bodyUnparser: Unparser)
   }
 }
 
-class ChoiceCombinatorUnparser(mgrd: ModelGroupRuntimeData, eventUnparserMap: Map[ChoiceBranchEvent, Unparser])
+class ChoiceCombinatorUnparser(
+  mgrd: ModelGroupRuntimeData,
+  eventUnparserMap: Map[ChoiceBranchEvent, Unparser],
+  choiceLengthKind: ChoiceLengthKind,
+  choiceLengthInBits: Long)
```

Comment:
I think if this becomes a ``Maybe[Long]`` then you don't need to pass in choiceLengthKind? So changing the type does simplify things a bit.